kusto/ingest/internal/filesystem/filesystem.go

```diff
@@ -362,3 +325,23 @@ func CompressionDiscovery(fName string) properties.CompressionType {
 	}
 	return properties.CTNone
 }
+
+var (
+	// gExtractURIProtocol is created outside the fucntion inorder to pay once for regexp creation.
```

Comment:
Not a fan of regexes, usually lead to something bad.  I'm an avoid if can.  Since we don't want to use a lexer/parser here, let's use url to our advantage:

https://play.golang.org/p/QplM9oEEGRm

```go
u, err := url.Parse(s)
if err != nil {
  return true
}
switch u.Scheme {
case "http", "https":
  return false
}
return true
```

Reply:
Using URL was what caused the problem in the first place.
Also, there are other schemes such as 'ftp' which are not file system, so your suggested implementation is inaccurate.

Reply:
You are right, but this leaves me liking this even less.

I don't think we handle file:// and I'm sure we don't handle "ftp://".  Its either blob or local file.  This of course is going to error somewhere farther down, but I'm not a huge fan of that.  

Filesystems, so many corner cases when you mix them.  Blah.

I don't want to enumerate every possible scheme, and because windows does c:\, it shows up as a url scheme.  
I also really, really don't want to do a lex/parser, cause that is a rabbit hole for little gain.

Maybe we just need to rethink this:  

```go
func IsLocalPath(s string) (bool, error) {
  u, err := url.Parse(s)
  if err != nil {
    return true
  }
  switch u.Scheme {
  // With this we know it SHOULD be a blobstore path.  It might not be, but I think that is a fine assumption to make.
  case "http", "https":
    return false, nil
  }

  // By this point, we know its not blobstore, so it needs to be something that gets resolved to a file.  
  // So we are going to Stat() the file and see if it exists and is not a directory.  
  // In your tests, this would fail "file://" which we don't support.  Also, because of this method, your tests
  // are going to be broken.  Can't test unix paths on windows, can't test windows path on unix.  
  // Best thing we can probably do is do tests with build tags.  Again, fileystems, blah....
  if stat, err := os.Stat(s); err != nil {
    return false, fmt.Errorf("is it a valid local file path(could not stat file) and was not a valid blob path")
  }
  if stat.IsDir() {
    return false, fmt.Errorf("path is a local directory and not a valid file")
  }
  return true, nil
}
```